task5/src/main/test/com/clouway/adapter/http/RegisterControllerTest.java

```diff
@@ -0,0 +1,120 @@
+package com.clouway.adapter.http;
+
+import com.clouway.core.User;
+import com.clouway.core.UserRepository;
+import com.clouway.core.Validator;
+import org.jmock.Expectations;
+import org.jmock.auto.Mock;
+import org.jmock.integration.junit4.JUnitRuleMockery;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Created by Kristiyan Petkov  <kristiqn.l.petkov@gmail.com> on 06.06.16.
+ */
+public class RegisterControllerTest {
+  private RegisterController registerController;
+
+  @Rule
+  public JUnitRuleMockery context = new JUnitRuleMockery();
+
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  @Mock
+  UserRepository userRepo;
+
+  @Mock
+  Validator dataValidator;
+
+  @Before
+  public void setUp() {
+    registerController = new RegisterController(userRepo, dataValidator);
+  }
+
+  @Test
+  public void happyPath() throws Exception {
+    final User user = new User("GlobalAdmin", "admin123456", "admin@clouwaybank.com");
+    final String email = user.getEmail();
+    final String password = user.getPassword();
+    final String username = user.getUsername();
+
+    context.checking(new Expectations() {{
+      oneOf(request).getParameter("regname");
```

Comment:
`username` instead of `regname`
and `password` is enought instead of `regpassword`